lab-davide/lib/client-pool.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+
+ // register socket(clients) to the pool
+function registerClient(socket, clientPool){
+  socket.wack = {};
+  socket.wack.id = 'user_' + Date.now();
+  clientPool.pool[socket.wack.id] = socket;
+}
+
+ // register event listeners to the new sockets(clients)
+function registerClientListeners(socket, clientPool){
+   // data event is triggered when ever the socket sends data
+
+   // data event is triggered when the socket(client) closes the connection
+  socket.on('close', function(data){
+   //per direction, delete from the clientpool property//
+    delete clientPool[socket.wack.id];
+    var message = socket.wack.id + ': ' + data.toString();
+    console.log(message + 'a client has disconnected');
```

Comment:
This might not be the most meaningful message you could write here. Also be careful about spaces--here, you would be missing a space between the massage and the word 'a'--an easy way to avoid that problem is to try using the es6 string methods.